Uchu.World/Systems/Missions/UseSkillTask.cs

```diff
@@ -12,17 +12,23 @@ public UseSkillTask(MissionInstance mission, int taskId, int missionTaskIndex)
 
         public override MissionTaskType Type => MissionTaskType.UseSkill;
 
-        public override bool Completed => Parameters.Any(t => Progress.Contains(t));
+        public override bool Completed => Progress.Count >= RequiredProgress;
 
-        public async Task ReportProgress(int skillId)
+        public async Task ReportProgress(int skillId, Lot? target = null)
         {
             if (!Parameters.Contains(skillId))
                 return;
 
+            // TODO: Check if target is correct, if applicable
+            // This is only used for the Spinjitzu Initiate achievements (1139 and 1140).
+            // They're supposed to progress when you smash one of the targets using the specified skill.
+            // if (Targets.Any(x => x != 0) && (target == null || !Targets.Contains((int) target)))
+            //     return;
```

Comment:
Is there a reason why this is commented out?

Reply:
I've left it commented out for now because there isn't a mechanism in place to actually pass the target to this function.
The UseSkillTask function gets called as soon as a skill is started, not when any behaviour inside the skill tree successfully smashes an object, and I couldn't find a clean way to implement this right now.
Having it commented out means players will complete achievements 1139 and 1140 by just using the skill the required amount of times (regardless of whether or not they kill Crux Prime enemies with it); uncommenting it will make the achievements impossible to complete.